int32 deltaX
int32 deltaY